solidity/contracts/bridge/Fraud.sol

```diff
@@ -329,58 +305,51 @@ library Frauds {
         emit FraudChallengeDefeatTimedOut(walletPubKeyHash, sighash);
     }
 
-    /// @notice Sets the new value for the `slashingAmount` parameter.
-    /// @param _newSlashingAmount the new value for `slashingAmount`
-    function setSlashingAmount(Data storage self, uint256 _newSlashingAmount)
-        external
-    {
-        self.slashingAmount = _newSlashingAmount;
-        emit FraudSlashingAmountUpdated(_newSlashingAmount);
-    }
+    /// @notice Unwraps the fraud challenge by verifying the given challenge
+    ///         and returns the UTXO key extracted from the preimage.
+    /// @param walletPublicKey The public key of the wallet in the uncompressed
+    ///        and unprefixed format (64 bytes)
+    /// @param preimage The preimage which produces sighash used to generate the
+    ///        ECDSA signature that is the subject of the fraud claim. It is a
+    ///        serialized subset of the transaction. The exact subset used as
+    ///        the preimage depends on the transaction input the signature is
+    ///        produced for. See BIP-143 for reference
+    /// @param witness Flag indicating whether the preimage was produced for a
+    ///        witness input. True for witness, false for non-witness input.
+    /// @return utxoKey UTXO key that identifies spent input.
+    /// @dev Requirements:
+    ///      - `walletPublicKey` and `sighash` calculated as `hash256(preimage)`
+    ///        must identify an open fraud challenge
+    ///      - the preimage must be a valid preimage of a transaction generated
+    ///        according to the protocol rules and already proved in the Bridge
+    function unwrapChallenge(
```

Comment:
I'm not sure if we really need the `unwrapChallenge` function after the refactor.
It duplicates some work that is also performed in `defeatFraudChallenge` (calculating sighs, calculating the challenge key, accessing the challenge).
I think we should just copy the contents of `unwrapChallenge` function into `Fraud.defeatFraudChallenge, remove the duplicate work and remove the `unwrapChallenge`.

Reply:
I love it! Please see e0c361f8c64db5fcd9993659105692d0bf7ab7d0.